src/lists/LList.es6

```diff
@@ -0,0 +1,197 @@
+          LList.isLList(toApp) ? toApp.map(LList.getData) : 
+          LList.convertToSeq(toApp).toArray()
+        )
+      )
+    );
+  }
+
+  /**
+   * Returns a new list, with copies of the old list's elements, pointed
+   * in reverse order
+   * @return {[type]} [description]
+   */
+  reverse(){
+    let reversed = [];
+    let unShiftToList = (element) => { reversed.unshift(element)}
+    this.map(LList.getData).forEach(unShiftToList);
+    return new LList(reversed);
+  }
+
+  /**
+   * Returns a new list, sans the current list's head.
+   * Uses tail-sharing.
+   * @return {[type]} [description]
+   */
+  removeHead() {
+    let notFirst = (node) => {
+      return (node !==  this.head);
+    }
+    return new LList(this.filter(notFirst).map(LList.getData));
+  }
+
+  /**
+   * Returns a new list in O(n), sans the current list's tail.
+   * @return {[type]} [description]
+   */
+  removeTail() {
+    let notLast = (node) => {
+      return (node !== this.tail)
+    }
+    return new LList(this.filter(notLast).map(LList.getData));
+  }
+
+  //Functional helper methods
+  forEach(cb) {
+    let current = this.head;
+    while (current !== null){
+      cb(current);
+      current = current.next;
+    }
+  }
+
+  map(cb){
+    let mapped = [];
+    let pushResult = (node) => { mapped.push(cb(node));}
+    this.forEach(pushResult);
+    return mapped;
+  }
+
+  filter(predicate) {
+    let filtered = [];
+    this.forEach((node) => {
+      if(!!predicate(node)){
+        filtered.push(node);
+      }
+    });
+    return filtered;
+  }
+
+  /**
+   * Creates individual nodes, from anything that can be stored
+   * in an immutable Seq.
+   * Can be passed null to create tails.
+   * @param  {[Primitive, Object]}   data []
+   * @param  {[New Node, null]} next []
+   * @return {[object]}        []
+   */
+  static makeNode(data, next) {
+    let node = {
+      data: data, 
+      next: next,
+    };
+    node[LList._LLNODE_SENTINEL_()] = true;
+    return node;
+  }
+
+  static makeHead(seq) {
+    if (seq === null || seq.size === 0) { 
+      return null; 
+    } else {
+      let rest = seq.rest();
+      rest = (rest.size === 0) ? null : rest;
+      return LList.makeNode(seq.first(), LList.makeHead(rest));
+    } 
+  }
+
+  //Extracts data from Nodes
+  static getData(node) {
+    return (
+      (LList.isNode(node)) ? node.data : 
+      new Error('getData only accepts nodes.')
+    );
+  }
+
+  static isLList(maybeLList){
+    return !!(maybeLList && maybeLList[LList._LL_SENTINEL_()]);
+  }
+
+  static isNode(maybeNode){
+    return !!(maybeNode && maybeNode[LList._LLNODE_SENTINEL_()]);
+  }
+
+  static convertToSeq(itemOrList){
+    return Array.isArray(itemOrList) ? IM.Seq(itemOrList) :
+      IM.Seq([].concat(itemOrList));
+  }
+
+  static _LL_SENTINEL_(){
+    return "@@__LINKED_LIST__@@"
+  }
+
+  static _LLNODE_SENTINEL_(){
+    return "@@__LL_NODE__@@"
+  }
+}
+
+LList.prototype[LList._LL_SENTINEL_()] = true;
```

Comment:
Note to self: 
-Implement circular with a sentinel node and/or count based on size
Perhaps easier to reason about after implementing DLL, since one can simply connect the nodes. It's up to what cannonical methods a circular list has...